MARINER.md

```diff
@@ -0,0 +1,78 @@
+
+More info about `apt` vs `tdnf` can be found [here](https://eng.ms/docs/products/mariner-linux/onboarding/packaging/packagemanagement).
+
+## Adding an RPM Repository
+New repositories are added with a `*.repo` file stored in the `/etc/yum.repos.d/` directory.
+
+Mariner by default has certain repo files in the container already. These include the repos for the mariner base and mariner extended packages.
+
+`azure-mdsd` is in the `Mariner-extras` repository.
+
+## Distroless Containers
+Distroless containers exclude packages provided by the distro such as a shell and package manager. They provide a slimmed down image with a smaller attack surface. We will still need a shell for our `main.sh` script and any debugging done by exec-ing into the container. But a distroless base image is still useful to not have the package managers and any unecessary packages which are needed solely for building and not at runtime. This is especially useful since `tdnf` does not have an `autoremove` command and using the distroless container trims down the size quite a bit.
+
+### Docker Multi-Stage Builds
+Since the distroless container does not have a package manager, we will still need a way to install all the necessary packages. The standard way to do this for distroless containers is through [multi-stage builds](https://docs.docker.com/develop/develop-images/multistage-build/). The first build uses the regular base container and sets up everything as normal up until adding the entry command as running the `main.sh` script.
+
+The second build uses the distroless image as the base image. Only the files we need at runtime are copied over from the first build. An example of this is [here](https://medium.com/@alexanto222/hardening-of-docker-images-distroless-images-d6d87b591a59).
+
+Note that this means that any new files added while developing will need to be copied over to the distroless build stage. Everything under `/opt` is currently copied over. See below for the steps to take when adding a new package.
+
+If you are building locally, you will need to set `DOCKER_BUILDKIT=1` because of [this](https://github.com/moby/moby/issues/37965) Docker bug. This has been set in our github actions builds.
+
+
+### Using the Shell in the Distroless Container
+`bash` is still copied over into our container to be able to run our `main.sh` bash script. However, there are some interface issues with `bash` on the distroless container. Mariner has default support for using `busybox` as the shell for debugging. This however changes some of the commands when exec-ing into the container which I have noted below. Feel free to add more here if you find them.
+
+  | Old Command | New Command |
+  | --- | --- |
+  | `ps -aux` | `ps` |
+
```

Comment:
is this because of the multiple containers in the pod and indicating defaulting to busybox ?

Reply:
No this is just an example of the command differences between busybox and bash